### PR TITLE
Fix post visibility regarding circles

### LINKED ISSUE #155: Bug: Fix public visibilty of posts

### PR DESCRIPTION
**Pull request by Gjum**
Replaces #165.

Closes #155.

### CODE REVIEW

File: posts/templates/posts/_post_create.html [hunk 1]
```diff
@@ -13,16 +13,17 @@
...
     <div class="row">
-        {% if user.circles %}
-            <div class="form-group col-xs-12 col-md-4">
+        <div class="form-group">
+            <div class="col-xs-12 col-md-4">
                 <select class="form-control" name="circle">
-                    <option disabled selected="selected">Select a circle</option>
-                    {% for circle in user.circles|dictsort:'name' %}
-                        <option value="{{ circle.pk }}">{{ circle.name }}</option>
+                    <option value="-2"{% if post.circles_id == -2 %} selected{% endif %}>Only visible to me</option>
```
Comment (juanecabellob) [inline posts/templates/posts/_post_create.html:19]: I would write 'Me' as its self-descriptive in this context.

Comment (Gjum) [inline posts/templates/posts/_post_create.html:19]: I used the word "visible" to create the context in the first place. How would a new user know what the drop down is for when it only says "Me" and "Public", and there is no label next to it?

Comment (juanecabellob) [inline posts/templates/posts/_post_create.html:19]: You're right with the context, but that doesn't sounds like a name for a circle. Maybe there is another way to create a context.

Comment (metalmatze) [inline posts/templates/posts/_post_create.html:19]: I would also be more specific. `Me` is to short. Facebook says `Only me` but why not `Only visible to me`. :wink:

File: circles/models.py [hunk 2]
```diff
@@ -2,6 +2,8 @@
 from django.db import models
 from users.models import User
 
+ME_CIRCLE = -2
+PUBLIC_CIRCLE = -1
```
Comment (metalmatze) [inline circles/models.py:6]: NICE!

File: posts/views.py [hunk 3]
```diff
@@ -71,31 +71,28 @@ def set_post_extra(post, request):
...
+    """
+    Returns a pre-filtered query object containing all posts visible for the specified user.
+    @param user: instance of the User model
+    """
+    own = Post.objects.filter(author=user)
+    public = Post.objects.filter(circles=PUBLIC_CIRCLE)
+    my_circle = Post.objects.filter(circles__owner=user.pk)
+    in_circle = Post.objects.filter(circles__members=user.pk)
+    return own | public | my_circle | in_circle
+
```
Comment (metalmatze) [inline posts/views.py:94]: This is awesome!

File: circles/models.py (hunk 2 above)
Comment (Gjum) [inline circles/models.py:6]: Depends. It's not so [nice](http://www.nice.org.uk/) that I cannot use these in the [template](https://github.com/7Pros/circuit/pull/182/files#diff-63d6d030a2c600c30247d13b745697feR20).

**Pull request merged**